buildSrc/subprojects/buildquality/src/main/kotlin/org/gradle/gradlebuild/buildquality/classycle/Classycle.kt

```diff
@@ -40,30 +40,21 @@ import javax.inject.Inject
 
 
 @CacheableTask
-open class Classycle : DefaultTask() {
-
-    @get:Internal
-    lateinit var classesDirs: FileCollection
+open class Classycle @Inject constructor(
+    @get:Internal val classesDirs: FileCollection,
+    @get:Input val excludePatterns: Provider<List<String>>,
+    @get:Input val reportName: String,
+    @get:Internal val reportDir: File,
+    @get:InputFile @get:PathSensitive(PathSensitivity.NONE) val reportResourcesZip: RegularFileProperty,
```

Comment:
❌ This should be a `Provider<RegularFile>` since we do not want this to be modified. The extension property should be modified, but not the task property.

Reply:
done. I had not done this because I was trying to avoid changing the logic in the `@TaskAction` as it was using `RegularFileProperty.getAsFile()`, but (as you'll see below) the change here didn't really cause much change below.